--- a/setuptools/depends.py.old	2019-06-20 16:02:37.763575776 +0200
+++ b/setuptools/depends.py	2019-06-20 16:03:25.849632795 +0200
@@ -1,5 +1,8 @@
 import sys
-import imp
+import warnings
+with warnings.catch_warnings():
+    warnings.simplefilter('ignore', DeprecationWarning)
+    import imp
 import marshal
 from distutils.version import StrictVersion
 from imp import PKG_DIRECTORY, PY_COMPILED, PY_SOURCE, PY_FROZEN
