--- a/pip/_vendor/msgpack/fallback.py.old	2020-03-20 18:02:37.755433719 +0100
+++ b/pip/_vendor/msgpack/fallback.py	2020-03-20 18:03:56.436490096 +0100
@@ -130,6 +130,9 @@
 
     See :class:`Unpacker` for options.
     """
+    if "encoding" in kwargs:
+        kwargs['raw'] = True
+        kwargs.pop('encoding')
     unpacker = Unpacker(None, max_buffer_size=len(packed), **kwargs)
     unpacker.feed(packed)
     try:
